R/reactives.R

```diff
@@ -2472,11 +2472,14 @@ debounce <- function(r, millis, priority = 100, domain = getDefaultReactiveDomai
 
       # Ensure r() is called only after setting firstRun to FALSE since r()
       # may throw an error
-      r()
+      try(r(), silent=TRUE)
       return()
     }
     # This ensures r() is still tracked after firstRun
-    r()
+    # TODO: If this fails, should we still update v$when?
+    # TODO: Does it make sense to update v$when before, rather
+    #       than after, executing r()?
```

Comment:
I'll start with the caveat that it's possible I'm misunderstanding the current behavior. That said...
I think it makes sense to reset the timer even in the case of an error. Since `debounce` will propagate errors from `r()`(I think), it makes sense to treat an execution of `r()` that throws error just like any other execution of `r()`.

Suppose there's a debouncer with a 3 second limit. Imagine `r()` fired once at time=0 with a good value, and then kept firing once per second for 4 more seconds, but threw errors each time, and then it fired once with a good value. It would be strange if, at time=3, the debouncer propagated the error, but `r()` kept executing each second.